File: challenges/3-tipos-de-triangulo/go/KMarculo/desafio3.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func main() {
	triangulo := make([]float64, 4)
	fmt.Println("Digite o tamanho dos tres lados do traingulo")
	fmt.Scan(&triangulo[0], &triangulo[1], &triangulo[2])
	fmt.Printf("Entrada: %.2f %.2f %.2f\n", triangulo[0], triangulo[1], triangulo[2])
	//Organiza os valores da slice de maior para menor
	sort.Sort(sort.Reverse(sort.Float64Slice(triangulo)))
	if triangulo[0] >= (triangulo[1] + triangulo[2]) {
		fmt.Printf("NAO FORMA TRIANGULO\n")
	} else {
		if (math.Pow(triangulo[0], 2)) == (math.Pow(triangulo[1], 2) + (math.Pow(triangulo[2], 2))) {
			fmt.Printf("TRIANGULO RETANGULO\n")
		}
		if (math.Pow(triangulo[0], 2)) > (math.Pow(triangulo[1], 2) + (math.Pow(triangulo[2], 2))) {
			fmt.Printf("TRIANGULO OBTUSANGULO\n")
		}
		if (math.Pow(triangulo[0], 2)) < (math.Pow(triangulo[1], 2) + (math.Pow(triangulo[2], 2))) {
			fmt.Printf("TRIANGULO ACUTANGULO\n")
		}
		if triangulo[0] == triangulo[1] && triangulo[1] == triangulo[2] {
			fmt.Printf("TRIANGULO EQUILATERO\n")
		} else if triangulo[0] == triangulo[1] || triangulo[1] == triangulo[2] || triangulo[2] == triangulo[0] {
			fmt.Printf("TRIANGULO ISOSCELES\n")
		}
	}
}

File: challenges/3-tipos-de-triangulo/python/juliax5/desafio_3.py
from typing import List

def normaliza(valores: List):
    valores.sort()

    A = valores[2]
    B = valores[1]
    C = valores[0]

    if A >= B + C:
        print('NAO FORMA TRIANGULO')
    elif A**2 == B**2 + C**2:
        print('TRIANGULO RETANGULO')
    elif A**2 > B**2 + C**2:
        print('TRIANGULO OBTUSANGULO')
    elif A**2 < B**2 + C**2:
        print('TRIANGULO ACUTANGULO')
    if A == B == C:
        print('TRIANGULO EQUILATERO')
    elif A == B or A == C or B == C:
        print('TRIANGULO ISOSCELES')

x = float(input('Valor decimal do primeiro lado: '))
y = float(input('Valor decimal do segundo lado: '))
z = float(input('Valor decimal do terceiro lado: '))

valores = [x, y, z]
normaliza(valores)
